### PR TITLE
Add unit tests for CategoryService

### LINKED ISSUE #80: Add unit test for CategoryService
The service class is suited for unit tests. This is a good place to start our tests.

### PR DESCRIPTION
**Pull request by helenahalldiniths**
This will fix #80

### CODE REVIEW

**Comment by helenahalldiniths** [comment]
**Some info:**
- This is more of a draft
- I have commented quite a lot since i'm bad at naming test-methods, feel free to propose other names.
- Since the class mostly is responsible for forwarding requests to the repository the tests feel quite simple, I'm wondering if there is something else that should be tested...?

**Some thoughts I got when working with theses tests (about the rest of the code):**
- The "error handling" of the update method should be moved to the service class.
- The delete method looks quite weird... first we send in the id, then we look for a catogory with said id, then we extract the id from said category and sends it to the "deleteById" method of the repository. Maybe insted we could just call the "deleteById" method and catch the error it throws if the id isn't valid and then throw our costume made exception?

**Comment by LordRekishi** [comment]
Will take a look at this once I've gotten a bit further with JMS

File: src/test/java/se/iths/crimedatabase/service/CategoryServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+class CategoryServiceTest {
+
+    @Mock
+    private CategoryRepository repository;
+    @InjectMocks
+    private CategoryService service;
+
+    // CREATE
+    // When we call the create-method with a category,
+    // said category should be passes to the save method of the repository
```
Comment (LordRekishi) [inline src/test/java/se/iths/crimedatabase/service/CategoryServiceTest.java]: The comments are helpful for reviewing, but I believe we should remove them for the final merge?
Well named test methods are enough, or we could use `@DisplayName("Name here")`

File: src/test/java/se/iths/crimedatabase/service/CategoryServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    }
+
+    // FIND ALL
+    // When we call the findAll-method a List of all categories should be returned.
+    @Test
+    void callingFindAllShouldReturnAllCategories() {
+        List<Category> categories = List.of(
+                new Category().setName("Theft").setId(1L),
+                new Category().setName("Burglary").setId(2L),
+                new Category().setName("shoplifting").setId(3L));
```
Comment (LordRekishi) [inline src/test/java/se/iths/crimedatabase/service/CategoryServiceTest.java]:
```suggestion
                new Category().setName("Shoplifting").setId(3L));
```
Let's change to a capital 'S' to correspond with the other categories and look better

File: src/test/java/se/iths/crimedatabase/service/CategoryServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        category.setName("Test").setId(validId);
+        when(repository.findById(validId)).thenReturn(Optional.of(category));
+        when(repository.findById(invalidId)).thenReturn(Optional.empty());
+
+        Optional<Category> expectedCategory = service.findById(validId);
+        Optional<Category> expectedEmpty = service.findById(invalidId);
+
+        assertThat(expectedCategory).isEqualTo(Optional.of(category));
+        assertThat(expectedEmpty).isNotPresent();
+    }
```
Comment (LordRekishi) [inline src/test/java/se/iths/crimedatabase/service/CategoryServiceTest.java:67]: Should we perhaps make this into two separate test methods?

File: src/test/java/se/iths/crimedatabase/service/CategoryServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        Category category = new Category().setName("Test").setId(validId);
+        when(repository.findById(invalidId)).thenReturn(Optional.empty());
+        when(repository.findById(validId)).thenReturn(Optional.of(category));
+
+        service.delete(validId);
+
+        verify(repository, times(1)).deleteById(validId);
+        verify(repository, times(0)).deleteById(invalidId);
+        assertThrows(EntityNotFoundException.class, () -> service.delete(invalidId));
+    }
```
Comment (LordRekishi) [inline src/test/java/se/iths/crimedatabase/service/CategoryServiceTest.java]: Same here? Two methods instead?

**Review by LordRekishi — DISMISSED** [review]
Looks really nice otherwise, and full coverage! Nice!

**Comment by helenahalldiniths** [comment]
I have updated the branch and made some small changes.

It looks like I have made changes to 22 files... This is because I have rebased the new commits onto this branch. If you checkout the code and compare it with the current main branch you can see that the only difference is the test class.

**Comment by LordRekishi** [comment]
I think we should try to use merge more than rebase. Merge main into our branch before pushing. That will solve all these commits doubling up.

**Pull request merged**